test/testsyncengine.cpp

```diff
@@ -586,6 +586,34 @@ private slots:
         QVERIFY(localFileExists("A/.hidden"));
         QVERIFY(fakeFolder.currentRemoteState().find("B/.hidden"));
     }
+
+    void testNoLocalEncoding()
+    {
+        auto utf8Locale = QTextCodec::codecForLocale();
+        if (utf8Locale->mibEnum() != 106) {
+            qWarning() << "Test only works for UTF8 locale";
+            return;
+        }
+
+        FakeFolder fakeFolder{ FileInfo::A12_B12_C12_S12() };
+        QVERIFY(fakeFolder.syncOnce());
+        QCOMPARE(fakeFolder.currentLocalState(), fakeFolder.currentRemoteState());
+
+        fakeFolder.remoteModifier().insert("A/tößt");
+        QVERIFY(fakeFolder.syncOnce());
+        QVERIFY(fakeFolder.currentLocalState().find("A/tößt"));
+
+        QTextCodec::setCodecForLocale(QTextCodec::codecForName("ASCII"));
+        QVERIFY(QTextCodec::codecForLocale()->mibEnum() == 3);
+
+        fakeFolder.remoteModifier().insert("B/tößt");
+        QVERIFY(fakeFolder.syncOnce());
+        QVERIFY(!fakeFolder.currentLocalState().find("B/tößt"));
+        QVERIFY(!fakeFolder.currentLocalState().find("B/t??t"));
+        QVERIFY(!fakeFolder.currentLocalState().find("B/t????t"));
```

Comment:
sync once more and check that the state on the server did not change. (no new files, no deleted files)